### PR TITLE
Add missing `border-radius` for `btn-group`

### LINKED ISSUE #34598: Button Group classes are missing border-radius properties
The Button Group (`btn-group`) classes are missing border-radius values. While the Buttons (`btn`) contained within the Button Group have their own border radius values, the Button Group itself also needs matching values. The problem is not apparent until a Shadow class is applied to a Button Group.

Applying a `shadow` or `shadow-lg` class to the `btn-group` produces a Button Group with a squared off white corners showing on all four corners of the Button Group, like so:

127210855-73e183d2-f5ca-448e-bf20-be52b7194919

To remedy this we just need to add border-radius values to the following classes:

- `btn-group` : 0.25rem
- `btn-group-sm` : 0.2rem
- `btn-group-lg` : 0.3rem

Doing so fixes the problem and eliminates the squared off corners of the `btn-group`, producing a Button group with the desired results:

CleanShot 2021-07-27 at 14 57 25@2x

### PR DESCRIPTION
**Pull request by AndyJiang99**
Fixes the border-radius for btn-group

Fixes #34598

### CODE REVIEW

**Review by ffoodd — CHANGES_REQUESTED** [review]
This should be added before the comment regarding double borders, with a line break between them.

Apart from that, I don't see any reason to reject this. Thanks!

**Comment by AndyJiang99** [comment]
> This should be added before the comment regarding double borders, with a line break between them.
> 
> Apart from that, I don't see any reason to reject this. Thanks!

Thanks for letting me know. I've addressed the changes, please take another look when you get a chance to. Thanks! Additionally, since I'm a first-time contributor for this repo, I need a maintainer to approve running workflows.

File: scss/_button-group.scss [hunk 1]
```diff
@@ -34,6 +34,9 @@
 }
 
 .btn-group {
+  // Prevent white corners if shadow or shadow-lg property applied
```
Comment (ffoodd) [inline scss/_button-group.scss:37]: This comment isn't really needed, the property and value speak for themselves. 

Then I guess it'd be good to go 👌

**Review by ffoodd — CHANGES_REQUESTED** [review]

**Comment by AndyJiang99** [comment]
Addressed linting issue for border-radius on the property-disallowed-list

**Review by ffoodd — CHANGES_REQUESTED** [review]

File: scss/_button-group.scss [hunk 2]
```diff
@@ -34,6 +34,8 @@
 }
 
 .btn-group {
+  border-radius: $btn-border-radius;
```
[the remark this answers is not in the document]
Comment (AndyJiang99) [reply scss/_button-group.scss]: Gotcha, understood! No worries, addressed the changes!

**Review by ffoodd — APPROVED** [review]
👌

**Pull request merged**